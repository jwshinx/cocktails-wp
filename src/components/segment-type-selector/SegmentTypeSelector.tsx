import React from 'react'

import {
  SegmentType,
  SegmentKeyType,
  FirstLetterSegmentKey,
  AlcoholicSegmentKey,
} from '../../models/cocktail'

interface SegmentTypeSelectorProps {
  onSegmentTypeChange: (value: SegmentType) => void
  onSegmentKeyChange: (value: SegmentKeyType) => void
  onSegmentValueChange: (value: string) => void
  onSortPropertyClick: (value: SegmentType) => void
  // onSortPropertyClick: (propertyType: PropertyType<T>) => void;
}

const SegmentTypeSelector = (props: SegmentTypeSelectorProps): JSX.Element => {
  const {
    onSegmentTypeChange,
    onSegmentKeyChange,
    onSegmentValueChange,
    onSortPropertyClick,
  } = props

  const onSelectChange = (value: string): void => {
    if (value === 'filter') {
      onSegmentKeyChange(AlcoholicSegmentKey)
    } else {
      onSegmentKeyChange(FirstLetterSegmentKey)
    }
    onSegmentTypeChange(value as SegmentType)
    onSegmentValueChange(AlcoholicSegmentKey.name)
    onSortPropertyClick(value as SegmentType)
  }

  return (
    <div className="form-group">
      <label htmlFor="type-selector">Enter type of search</label>
      <select
        className="form-select mb-3"
        id="type-selector"
        title="type-selector"
        // onChange={(event) => onSelectChange(event.target.value as keyof T)}
        onChange={(event) => onSelectChange(event.target.value)}
      >
        {['search', 'filter'].map((segmentType): JSX.Element => {
          return (
            <React.Fragment key={segmentType}>
              <option key={segmentType} value={segmentType}>
                {segmentType}
              </option>
            </React.Fragment>
          )
        })}
      </select>
    </div>
  )
}

export default SegmentTypeSelector
