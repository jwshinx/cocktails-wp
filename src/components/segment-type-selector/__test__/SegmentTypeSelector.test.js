/**
 * @jest-environment jsdom
 */

import React from 'react'
// import ReactDOM from 'react-dom'
// import { render } from '@testing-library/react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import SegmentTypeSelector from '../SegmentTypeSelector'

it('should render label "Enter category of search"', () => {
  render(<SegmentTypeSelector />)
  const elem = screen.getByText(/Enter type of search/)
  expect(elem).toBeInTheDocument()
})

it('should render combobox', () => {
  render(<SegmentTypeSelector />)
  const elem = screen.getByRole('combobox')
  expect(elem).toBeInTheDocument()
})

it('should render options "filter" and "search"', () => {
  render(<SegmentTypeSelector />)
  expect(screen.getAllByRole('option', { name: 'search' })).toBeTruthy()
  expect(screen.getAllByRole('option', { name: 'filter' })).toBeTruthy()
})

it('should snapshot SegmentTypeSelector', () => {
  const tree = renderer.create(<SegmentTypeSelector />).toJSON()
  expect(tree).toMatchSnapshot()
})
