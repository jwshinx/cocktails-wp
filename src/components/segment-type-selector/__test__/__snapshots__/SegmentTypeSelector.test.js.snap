// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot SegmentTypeSelector 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="sorter"
  >
    Enter type of search
  </label>
  <select
    className="form-select mb-3"
    id="sorter"
    onChange={[Function]}
  >
    <option
      value="search"
    >
      search
    </option>
    <option
      value="filter"
    >
      filter
    </option>
  </select>
</div>
`;
